at.medevit.elexis.emediplan.ui/src/at/medevit/elexis/emediplan/ui/ImportEMediplanDialog.java

```diff
@@ -0,0 +1,368 @@
+import org.eclipse.jface.dialogs.TitleAreaDialog;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.jface.viewers.ArrayContentProvider;
+import org.eclipse.jface.viewers.ColumnLabelProvider;
+import org.eclipse.jface.viewers.ColumnViewerToolTipSupport;
+import org.eclipse.jface.viewers.StructuredSelection;
+import org.eclipse.jface.viewers.TableViewer;
+import org.eclipse.jface.viewers.TableViewerColumn;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Menu;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.swt.widgets.TableColumn;
+
+import at.medevit.elexis.emediplan.core.model.chmed16a.Medicament;
+import at.medevit.elexis.emediplan.core.model.chmed16a.Medication;
+import ch.artikelstamm.elexis.common.ArtikelstammItem;
+import ch.elexis.core.data.events.ElexisEvent;
+import ch.elexis.core.data.events.ElexisEventDispatcher;
+import ch.elexis.core.data.events.ElexisEventListener;
+import ch.elexis.core.model.prescription.EntryType;
+import ch.elexis.core.ui.UiDesk;
+import ch.elexis.core.ui.events.ElexisUiEventListenerImpl;
+import ch.elexis.data.Patient;
+import ch.elexis.data.PersistentObject;
+import ch.elexis.data.Prescription;
+import ch.elexis.data.Query;
+import ch.rgw.tools.TimeTool;
+
+public class ImportEMediplanDialog extends TitleAreaDialog {
+	private final Medication medication;
+	
+	private TableViewer tableViewer;
+	private Table table;
+	
+	private ElexisEventListener eeli_presc = new ElexisUiEventListenerImpl(Prescription.class,
+		ElexisEvent.EVENT_CREATE | ElexisEvent.EVENT_UPDATE | ElexisEvent.EVENT_DELETE) {
+		public void runInUi(ElexisEvent ev){
+			if (medication.Patient != null) {
+				Prescription prescription = (Prescription) ev.getObject();
+				if (prescription != null && prescription.getArtikel() instanceof ArtikelstammItem) {
+					String patientId = prescription.get(Prescription.FLD_PATIENT_ID);
+					if (patientId.equals(medication.Patient.patientId))
+						for (Medicament medicament : medication.Medicaments) {
+							if (prescription.getArtikel().equals(medicament.artikelstammItem)) {
+								if ((ev.getType() == ElexisEvent.EVENT_CREATE
+									|| ev.getType() == ElexisEvent.EVENT_UPDATE)
+									&& prescription.getDosis().equals(medicament.dosis)) {
+									medicament.exists = true;
+									medicament.artikelstammItem =
+										(ArtikelstammItem) prescription.getArtikel();
+								} else if (ev.getType() == ElexisEvent.EVENT_DELETE
+									|| ev.getType() == ElexisEvent.EVENT_UPDATE) {
+									if (findPresciptionByMedicament(medicament).size() == 0) {
+										medicament.exists = false;
+									}
+									
+								}
+							}
```

Comment:
Isn't this the same functionality as the findExistingPrescription from the comment above? Can we reuse the code of the findPrescriptionByMedicament method?